cmake/find/FindJulia.cmake

```
@@ -0,0 +1,108 @@
+macro(julia_bail_if_false message var)
+  if(NOT ${var})
+    set(Julia_FOUND 0)
+    set(Julia_NOT_FOUND_MESSAGE "${message}")
+    return()
+  endif()
+endmacro()
+
+# https://docs.julialang.org/en/v1/manual/environment-variables/#JULIA_BINDIR
+find_program(JULIA_EXECUTABLE julia PATHS ENV JULIA_BINDIR)
+julia_bail_if_false("The Julia executable could not be found" JULIA_EXECUTABLE)
+
+if(NOT DEFINED JULIA_BINDIR)
+  # The executable could be a chocolatey shim, so run some Julia code to report
+  # the path of the BINDIR
+  execute_process(
+      COMMAND "${JULIA_EXECUTABLE}" -e "print(Sys.BINDIR)"
+      OUTPUT_VARIABLE julia_bindir
+  )
+  file(TO_CMAKE_PATH "${julia_bindir}" julia_bindir)
+  set(JULIA_BINDIR "${julia_bindir}" CACHE PATH "")
+endif()
+julia_bail_if_false("The Julia executable could not report its location" JULIA_BINDIR)
+
+get_filename_component(julia_prefix "${JULIA_BINDIR}" DIRECTORY)
+
+if(WIN32)
+  set(julia_old_CMAKE_FIND_LIBRARY_SUFFIXES "")
+  set(julia_old_CMAKE_FIND_LIBRARY_PREFIXES "")
+  if(CMAKE_FIND_LIBRARY_SUFFIXES)
+    set(julia_old_CMAKE_FIND_LIBRARY_SUFFIXES "${CMAKE_FIND_LIBRARY_SUFFIXES}")
+    list(APPEND CMAKE_FIND_LIBRARY_SUFFIXES .dll.a)
+  else()
+    set(CMAKE_FIND_LIBRARY_SUFFIXES ".lib;.dll.a")
+  endif()
+  if(CMAKE_FIND_LIBRARY_PREFIXES)
+    set(julia_old_CMAKE_FIND_LIBRARY_PREFIXES "${CMAKE_FIND_LIBRARY_PREFIXES}")
+    list(APPEND CMAKE_FIND_LIBRARY_PREFIXES lib)
+  else()
+    set(CMAKE_FIND_LIBRARY_PREFIXES ";lib")
+  endif()
+endif()
+
+find_library(JULIA_LIBRARY julia HINTS "${julia_prefix}/lib")
```

```suggestion
```suggestion
find_library(JULIA_LIBRARY julia HINTS "${julia_prefix}/lib")
if (NOT JULIA_LIBRARY)
    message(WARNING "Cannot find julia library in ${JULIA_PATH_PREFIX}/lib. Make sure JULIA_BINDIR is set correctly and that your image is uncompressed.")
endif()
```